src/leer.java

```diff
@@ -8,264 +8,210 @@
-		           id_ride = campos[0];
-		           lat = campos[1];
-		           lng = campos[2];
-		           timestamp = campos[3];
-		          
-		           // read next line 
-		           linea = bufferLectura.readLine();
-		   
-		           if (linea == null) {
+		           String[] fields_line1 = line.split(",");
+				   id_ride = fields_line1[0];
+		           lat = fields_line1[1];
+		           lng = fields_line1[2];
+		           timestamp = fields_line1[3];
+				   line = buffer_read.readLine();
+
+		           if (line == null) {
 			           break;
 		           }
 		   
-		           String[] campos2 = linea.split(",");
-		               
-		           lat1 = campos2[1];
-		           lng1 = campos2[2];
-		           timestamp1 = campos2[3];
-		           //calling calculating distance
-		           U=calculate.haversine(lat, lng, lat1, lng1, id_ride, timestamp, timestamp1,i);
-				   //calling filter
-		           campos=filter(campos,U);	   
-				   //System.out.println(Arrays.toString(campos));
-				   //System.out.println(U);
-				   
+		           String[] fields_line2 = line.split(",");
+		           lat_line2 = fields_line2[1];
+		           lng_line2 = fields_line2[2];
+		           timestamp_line2 = fields_line2[3];
+		           U=calculate.calculate_speed(lat, lng, lat_line2, lng_line2, id_ride, timestamp, timestamp_line2,i);
+				   filter(fields_line1,U);
 				   i++;
-	   }   
-	 }
-	 catch (IOException e) {
-	  e.printStackTrace();
-	 }
-	 finally {
-		 //calling valor to print answer
-	  result= calculate.valor(i,id_ridef,fare);    
-		 
-	  if (bufferLectura != null) {
-	   try {
-	    bufferLectura.close();
-	   } 
-	   
-	   catch (IOException e) {
-	    e.printStackTrace();
-	   }
-	  }
-	 
-	  try {
-          String ruta = "/Users/a.manning/Downloads/result.txt";
-          String contenido = result;
-            
-          File file = new File(ruta);
-          // if the file doesn't exists is created
-          if (!file.exists()) {
-              file.createNewFile();
-          }
-          FileWriter fw = new FileWriter(file);
-          BufferedWriter bw = new BufferedWriter(fw);
-          
-		bw.write(contenido);
-          bw.close();
-      } catch (Exception e) {
-          e.printStackTrace();
-      }
-  }
-	 } 
-     	 	
-
-
-
-
-public  double haversine (String lat,String lng, String lat1,String lng1, String id_ride, String timestamp, String timestamp1,int i){
- 
-	double a,c,distance,U,deltlat,deltlng,delttime;
+	            }
+	    }
+	    catch (IOException e) {
+	          e.printStackTrace();
+	    }
+	    finally {
+	        result= calculate.prepare_result(i,id_ride_final,fare);
+		    if (buffer_read != null) {
+	           try {
+	               buffer_read.close();
+	           }
+	           catch (IOException e) {
+	                e.printStackTrace();
+			   }
+	        }
+		    try {
+                String route = "/Users/a.manning/Downloads/result.txt";
```

Comment:
"....result.txt" is a magic number. Is something that you can put in a constant or read from env variables.